Game/MyProject/Game/Camera.hpp

```diff
@@ -1,60 +1,80 @@
 #pragma once
 #include <stdio.h>
 #include <IvVector3.h>
+#include <IvVector2.h>
+#include <IvVector4.h>
 #include <IvMatrix44.h>
 #include <IvMatrix33.h>
 #include <IvMath.h>
+#include <iostream>
 
 class Camera{
 
 private:
     
     IvMatrix44 transform;
+    IvMatrix44 rotMatrix;
     
     IvVector3 position;
     IvVector3 lookAt;
-    IvVector3 orientation;
+    IvVector3 rotation;
+    IvVector3 rotOffsets;
     
     float fieldOfView;
-    float near;
-    float far;
-    float aspectRatio;
-    int width;
-    int height;
+    float nearPlane;
+    float farPlane;
+    int screenWidth;
+    int screenHeight;
 
-    
 public:
     Camera(){};
-    Camera(float FOV, float nearPlane, float farPlane, int width, int height);
+    Camera(const float fieldOfView, const float nearPlane, const float farPlane, const int screenWidth, const int screenHeight) :
+        rotOffsets(0.0f, 0.0f, 0.0f),
+        fieldOfView(fieldOfView),
+        nearPlane(nearPlane),
+        farPlane(farPlane),
+        screenWidth(screenWidth),
+        screenHeight(screenHeight)
+        {};
+    
     ~Camera();
     
-    void setPosition(IvVector3 newPos);
-    void setLookAt(IvVector3 newLookAt);
-    void setOrientation(IvVector3 newOrientation);
+    void setPosition(const IvVector3& newPos);
+    void setLookAt(const IvVector3& newLookAt);
+    void setRotation(const IvVector3& newRotation);
     void setFieldOfView(float newFOV);
     void setNear(float newNear);
     void setFar(float newFar);
+    void setScreenWidth(float newWidth);
+    void setScreenHeight(float newHeight);
+    
+    const float getFieldOfView() { return fieldOfView; }
```

Comment:
You are returning const values from non-const functions. Mark the getters as const member functions.
https://www.geeksforgeeks.org/const-member-functions-c/

Reply:
Also, this applies for multiple files. Check all occurences.